c2corg_ui/views/document.py

```diff
@@ -413,20 +421,43 @@ def _get_api_cache_key_from_etag(self, etag):
     def _get_response(self, page_html):
         return get_response(self.request, page_html)
 
+    def _redirect(self, id, lang, slug=None):
+        location = ''
+        if slug is None:
+            location = self.request.route_url(
+                self._API_ROUTE + '_view_id_lang', id=id, lang=lang)
+        else:
+            location = self.request.route_url(
+                self._API_ROUTE + '_view', id=id, lang=lang, slug=slug)
+        raise HTTPMovedPermanently(location=location)
```

Comment:
> Why would it change? This case has never happened in v5 since 2007.

How did you check this? Our model does not prevent that `redirects_to` can not be changed anymore if it has been set once.

> My concern is that we need to make sure that redirecting documents are not referenced by search engines and that the target document is always taken into accound instead. It seems we need the 301 for this, doesn't it?

The redirected page will also be indexed with 302. This graphic explains it quite well: https://seohackercdn-seohacker.netdna-ssl.com/wp-content/uploads/2011/10/301-and-302-Redirect-small2.png